jib-core/src/main/java/com/google/cloud/tools/jib/event/progress/ThrottledLongConsumer.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2019 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.event.progress;
+
+import java.io.Closeable;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.function.Consumer;
+import java.util.function.Supplier;
+import javax.annotation.Nullable;
+
+/**
+ * Wraps a {@code Consumer<Long>} so that multiple consume calls ({@link #accept}) within a short
+ * period of time are merged into a single later call with the value accumulated up to that point.
+ */
+public class ThrottledLongConsumer implements Consumer<Long>, Closeable {
```

Comment:
ThrottledAccumulator or ThrottledAccumulatingConsumer?

Reply:
Going with `ThrottledAccumulatingConsumer`, as I want it to show that it's fundamentally a `Consumer`.